### PR TITLE
Add  `base_dir` kwarg to bias_steps.run_analysis

### PR DESCRIPTION
**Pull request by dpdutcher**
Addresses #214 

Tested at Princeton.

### CODE REVIEW

File: sodetlib/operations/bias_steps.py [hunk 1]
```diff
@@ -467,7 +470,8 @@ def _load_am(self, arc=None, fix_timestamps=True):
             if arc:
                 self.am = arc.load_data(self.start, self.stop)
             else:
-                self.am = sdl.load_session(self.meta['stream_id'], self.sid)
+                self.am = sdl.load_session(self.meta['stream_id'], self.sid,
```
Comment (msilvafe) [inline sodetlib/operations/bias_steps.py:473]: It doesn't look like `load_session` can currently take a `base_dir` argument. Also, out of curiosity, what issue is this PR addressing?

Comment (dpdutcher) [reply sodetlib/operations/bias_steps.py:473]: https://github.com/simonsobs/sodetlib/blob/4aafb94bb36cac85b754b0a1e28575d33e1b6243/sodetlib/stream.py#L71-L72

Comment (msilvafe) [reply sodetlib/operations/bias_steps.py:473]: Interesting...I was looking at `load_session` here: https://github.com/simonsobs/sodetlib/blob/4aafb94bb36cac85b754b0a1e28575d33e1b6243/sodetlib/legacy/smurf_funcs/smurf_ops.py#L422 not sure why we have a duplicate.

**Review by jlashner — APPROVED** [review]
Looks good to me! I don't love hardcoding `/data/so/timestreams` but I take full responsibility for that bc how  `load_session` is implemented, and recognize there is not really an easy robust way to default to `cfg.g3_dir`

File: sodetlib/operations/bias_steps.py (hunk 1 above)
Comment (msilvafe) [reply sodetlib/operations/bias_steps.py:473]: Ahh it's in "legacy" just searched in github and it gave me that one. Ok, this all makes sense we should just accept this PR probably but I'll wait an hr or so for @jlashner to have a chance to take a look.

Comment (dpdutcher) [reply sodetlib/operations/bias_steps.py:473]: I linked Issue #214 , which has a slightly different title, but the key aspect there being there was no base_dir kwarg being passed to the load_session call in the bias step analysis.

This problem arises when trying to analyze bias steps without an G3tSMuRF archive on a machine other than the smurf server. For example, my timestreams on simons1 are not located at /data/so/timestreams.

**Comment by dpdutcher** [comment]
Yeah my first instinct was to go with default `base_dir=None`, but that kind of clashed with how it was implemented elsewhere, and I touch less files this way.

**Pull request merged**